Sources/Flightdeck/Flightdeck.swift

```diff
@@ -1,6 +1,416 @@
+        
+    }
+    // MARK: - setSuperProperties
+    
+    /**
+     Sets properties that are included with each event during the duration of the current initialization.
+     Super properties are reset everytime the app is terminated. Make sure to set necessary super properties everytime after Flightdeck.initialize() is called.
+     
+     Super properties can be overwritten by similarly named properties that are provided with trackEvent()
+     
+     - parameter properties: properties dictionary
+     */
+    
+    public func setSuperProperties(_ properties: [String: Any]) {
+        self.superProperties = properties
+    }
+    
+    
+    
+    // MARK: - trackEvent
+    
+    /**
+     Tracks an event with properties.
+     Properties are optional and can be added only if needed.
+     
+     Properties will allow you to segment your events in your Mixpanel reports.
+     Property keys must be String objects and the supported value types need to conform to MixpanelType.
+     MixpanelType can be either String, Int, UInt, Double, Float, Bool, [MixpanelType], [String: MixpanelType], Date, URL, or NSNull.
+     If the event is being timed, the timer will stop and be added as a property.
+     
+     - parameter event:      event name
+     - parameter properties: properties dictionary
+     */
+    
+    /// Public trackEvent function
+    public func trackEvent(_ event: String, properties: [String: Any]? = nil){
+        if event.hasPrefix(self.automaticEventsPrefix) {
+            logger.error("Flightdeck: Event name has forbidden prefix \(self.automaticEventsPrefix)")
+        } else {
+            self.trackEventCore(event, properties: properties)
+        }
+    }
+    
+    /// Private trackEvent function used for automatic events
+    private func trackAutomaticEvent(_ event: String, properties: [String: Any]? = nil){
+        if self.trackAutomaticEvents {
+            self.trackEventCore("\(self.automaticEventsPrefix)\(event)", properties: properties)
+        }
+    }
+    
+    /// Private trackEventCore function
+    private func trackEventCore(_ event: String, properties: [String: Any]? = nil){
+        
+        let currentDateTime = self.getCurrentDateTime()
+        
+        /// Initialize a new Event object with event string and current UTC datetime
+        var eventData = Event(
+            clientType: self.clientType,
+            clientVersion: self.clientVersion,
+            clientConfig: self.clientConfig,
+            event: event,
+            datetimeUTC: currentDateTime.datetimeUTC
+        )
+        
+        /// Set custom properties, merged with super properties, if any
+        if var props = properties {
+            props.merge(self.superProperties) { (current, _) in current }
+            eventData.properties = self.stringifyProperties(properties: props)
+        } else if !self.superProperties.isEmpty {
+            eventData.properties = self.stringifyProperties(properties: superProperties)
```

Comment:
why json-encode them here already instead of when the entire Event is encoded (before sending)?

Reply:
if you make your superproperties non-optional, you could save a few lines here: 
```
var allProperties = superProperties ?? [:]
if properties
{
    allProperties.merge(properties)
}
// do something with allProperties
allProperties.stringify()
``